boards/espressif/esp32c6_devkitc/doc/index.rst

```diff
@@ -0,0 +1,264 @@
+It has a 320KB ROM, a 512KB SRAM, and works with external flash. [1]_
+
+ESP32-C6-DevKitC-1 is an entry-level development board based on ESP32-C6-WROOM-1(U),
+a general-purpose module with a 8 MB SPI flash.
+
+Most of the I/O pins are broken out to the pin headers on both sides for easy interfacing.
+Developers can either connect peripherals with jumper wires or mount ESP32-C6-DevKitC-1 on
+a breadboard. [2]_
+
+ESP32-C6 includes the following features:
+
+- 32-bit core RISC-V microcontroller with a clock speed of up to 160 MHz
+- 400 KB of internal RAM
+- WiFi 802.11 ax 2.4GHz
+- Fully compatible with IEEE 802.11b/g/n protocol
+- Bluetooth LE: Bluetooth 5.3 certified
+- Internal co-existence mechanism between Wi-Fi and Bluetooth to share the same antenna
+- IEEE 802.15.4 (Zigbee and Thread)
+
+Digital interfaces:
+
+- 30x GPIOs (QFN40), or 22x GPIOs (QFN32)
+- 2x UART
+- 1x Low-power (LP) UART
+- 1x General purpose SPI
+- 1x I2C
+- 1x Low-power (LP) I2C
+- 1x I2S
+- 1x Pulse counter
+- 1x USB Serial/JTAG controller
+- 1x TWAI® controller, compatible with ISO 11898-1 (CAN Specification 2.0)
+- 1x SDIO 2.0 slave controller
+- LED PWM controller, up to 6 channels
+- 1x Motor control PWM (MCPWM)
+- 1x Remote control peripehral
+- 1x Parallel IO interface (PARLIO)
+- General DMA controller (GDMA), with 3 transmit channels and 3 receive channels
+- Event task matrix (ETM)
+
+Analog interfaces:
+
+- 1x 12-bit SAR ADCs, up to 7 channels
+- 1x temperature sensor
+
+Timers:
+
+- 1x 52-bit system timer
+- 1x 54-bit general-purpose timers
+- 3x Watchdog timers
+- 1x Analog watchdog timer
+
+Low Power:
+
+- Four power modes designed for typical scenarios: Active, Modem-sleep, Light-sleep, Deep-sleep
+
+Security:
+
+- Secure boot
+- Flash encryption
+- 4-Kbit OTP, up to 1792 bits for users
+- Cryptographic hardware acceleration: (AES-128/256, ECC, HMAC, RSA, SHA, Digital signature, Hash)
+- Random number generator (RNG)
+
+For more information, check the datasheet at :ref:`ESP32-C6 Datasheet`_.
+
+Supported Features
+==================
+
+Current Zephyr's ESP32-C6-DevKitC board supports the following features:
+
++------------+------------+-------------------------------------+
+| Interface  | Controller | Driver/Component                    |
++============+============+=====================================+
++------------+------------+-------------------------------------+
+| UART       | on-chip    | serial port                         |
++------------+------------+-------------------------------------+
+| GPIO       | on-chip    | gpio                                |
++------------+------------+-------------------------------------+
+| PINMUX     | on-chip    | pinmux                              |
++------------+------------+-------------------------------------+
+| USB-JTAG   | on-chip    | hardware interface                  |
++------------+------------+-------------------------------------+
+
```

Comment:
In case ledc/pwm, interrupt, spi, pinctrl, timer, watchdog and are supported, please, include it here as well.

Reply:
@raffarost, please, update this entry as needed.